src/UCTSearch.cpp

```diff
@@ -522,12 +521,18 @@ int UCTSearch::get_best_move(passflag_t passflag) {
     return bestmove;
 }
 
-std::string UCTSearch::get_pv(FastState & state, UCTNode& parent) {
+std::string UCTSearch::get_pv(GameState & state, UCTNode& parent) {
     if (!parent.has_children()) {
         return std::string();
     }
 
+    float eval;
+    // we do this anyway even if we have children.  We don't want to crash the run
+    // just because the previonsly-expanded node is in the INITiAL state.
+    parent.create_children(m_network, m_nodes, state, eval, get_min_psa_ratio());
```

Comment:
Bah!

What about changing wait_expanded to bail before the assert if the node is in INITIAL?

Reply:
This is really tricky. get_pv wants to traverse m_children during the search so it really needs to wait until the node is fully expanded, or give up. Can we make it bail if any of the nodes it encounters is in that state? Together with your previous batch to not reset fully expanded nodes, this should work? PVs will be truncated shorter but that is OK.